sandbox/extract-imagery-metadata_mg.R

```diff
@@ -0,0 +1,181 @@
+# your 'extract_metadata_<initials>' function. Once it is in there, then you can run the top part of
+# this script again, and when it extracts the metadata for each EXIF dataset, your additional
+# metadata should be included.
+
+
+#START CODE FOR CENTROID LAT/LONG
+
+GPSLocation = exif$GPSPosition
+
+#splitting the list (was this step really necessary for my dataset? I'm not really sure this actually helped?)
+coordinates <- strsplit(GPSLocation, split = ";")
+
+# Separating the GPS from the whole exif file
+almostdone <- strsplit(unlist(coordinates), split = ",")
+
+#Creating a seperate dataframe with just the coordinates to call upon later
+isolatedcoordinates <- as.data.frame(do.call(rbind, almostdone))
+
+library(stringr)
+#Need to download this package as it will help with the following functions
+separator(df)
+
+install.packages("tidyverse")
+library(tidyverse)
+
+df2 = separate(exif,
+        col = GPSPosition,
+        into = c("lat", "long"),
+        sep = " " )
+#Here this worked, instead of doing by number of characters, did it by seperating at
+#the space in between the two (since the number of characters differed from row to row).
+
+print(df2)
+#printed out the new df that includes the "lat" and "long"
+
+#when looking online, it says that the way to find the centroid lat/long is to
+#just use the average of the lat and the average of the long
+
+str(df2$lat)
+str(df2$long)
+
+#here checking what type of character the lat and long are within the
+#dataset, if it is not numeric, will need to change it.
+
+df2$lat = as.numeric(df2$lat)
+df2$long = as.numeric(df2$long)
+
+#had to convert to numeric numbers since it was thinking that the column was a
+#character.
+
+xcoordinate = mean(df2$lat, na.rm=TRUE)
+ycoordinate = mean(df2$long, na.rm= TRUE)
+
+#finding the mean of the lat and long, since this is all one mission,
+#it should give us the centroid lat/long, at least from what I researched.
+
+print(ycoordinate)
+print(xcoordinate)
+
+#END CODE FOR CENTOID CALCULATIONS
+
+
+
+#START CODE FOR SOLAR NOON CALCULATIONS
+
+#found the package suntools which will calculate the solar noon, this way we do not have to do
+#any crazy calculations, it is all included in the function and only needs lat/long and the time.
+
+install.packages("suntools")
+library("suntools")
+
+#Here we are seperating the date from the time, since we only need the date to run this function
+datetime <- df2[1, "capture_datetime"]
+datetime
+datetimedf = separate(exif,
+               col = capture_datetime,
+               into = c("date", "time"),
+               sep = " " )
+datetimedf
+date <- datetimedf[1, 143]$date
+date
+#now using the newly seperated date column and using the first row to get the date to be used in the function below
+
+#used the centroid calculated in the above code chunk for the lat/long
+solarnoon <- solarnoon(
+  matrix(c(xcoordinate, ycoordinate), nrow = 1),
+  as.POSIXct(date),
+  POSIXct.out=TRUE
+)
```

Comment:
When I ran it, this function said solarnoon was 2 am :P . I tested what happened when I swapped the x and y coords, and it gave a reasonable result. Apparently it expects the reverse order. I didn't see this in the documentation anywhere though!

Reply:
I think that `suntools::solarnoon()` does require x, then y, but I think the original extraction of those coordinates using `tidyr::separate()` earlier in the code assigned latitude to `xcoordinate` and longitude to `ycoordinate` so they were passed backwards here in L166 even though the labeling is correct. Another good reason to take Derek's suggestion and calculate the centroid of all of the lon/lat and then pass the centroid as an {sf} object. Something like:

```
exif_centroid = sf::st_union(exif) |> sf::st_centroid()
```

`sf::st_union()` takes all the individual points and collapses them into a single MULTIPOINT. Then the `sf::st_centroid()` function takes the mean of all the latitudes and the mean of all the longitudes (as you discovered) to calculate the centroid.

Does the `suntools::solarnoon()` function expect that the coordinate reference system (CRS) of the {sf} object is something specific? If so, you might want to use `sf::st_transform()` to transform the centroid to the expected CRS. But maybe the `suntools()` function does that for you!

If I had a nickel for every time I switched lon/lat and x/y...

You might also consider assigning the result of the `suntools::solarnoon()` function call to something other than `solarnoon` since you are overwriting the `suntools::solarnoon()` function with the calculation! Like `exif_solarnoon` or `flight_solarnoon` (whatever the unit is represented by the `exif` object.